pkg/kubelet/kubelet.go

```diff
@@ -685,7 +686,7 @@ func NewMainKubelet(kubeCfg *kubeletconfiginternal.KubeletConfiguration,
 		// be removed. Related issue:
 		// https://github.com/kubernetes/kubernetes/issues/51798
 		if (kubeCfg.ContainerRuntime == kubetypes.DockerContainerRuntime &&
-			goruntime.GOOS == "linux") || remoteRuntimeEndpoint == "/var/run/crio.sock" {
+			goruntime.GOOS == "linux") || remoteRuntimeEndpoint == cadvisorcrio.CrioSocket {
```

Comment:
can we just have this check if remoteRuntimeEndpoint has the suffix crio.sock so it can be anywhere and still work?

Reply:
then leave a // TODO for when we next pick cAdvisor to update?

Reply:
Is the crio.sock location hardcoded in CAdvisor? Wouldn't remoteRuntimeEndpoint need to exactly match CAdvisor's expectation in that case?